File: app-mvp/src/main/java/com/xuie/imaginaryandroid/gui/nets/video/VideosContract.java
package xuk.imaginary.gui.nets.video;

import xuk.imaginary.BasePresenter;
import xuk.imaginary.BaseView;
import xuk.imaginary.data.VideoBean;

import java.util.List;

/**
 * This specifies the contract between the view and the presenter.
 *
 * @author xuie
 */
public interface VideosContract {

    interface View extends BaseView<Presenter> {

        void addList(boolean isRefresh, List<VideoBean> videoBeen);
    }

    interface Presenter extends BasePresenter {
        void getList(String type, boolean isRefresh);
    }
}

File: app-mvp-kotlin/src/main/java/xuk/imaginary/gui/nets/video/VideosContract.kt
package xuk.imaginary.gui.nets.video

import xuk.imaginary.BasePresenter
import xuk.imaginary.BaseView
import xuk.imaginary.data.VideoBean

/**
 * This specifies the contract between the view and the presenter.
 */
interface VideosContract {

  interface View : BaseView<Presenter> {

    fun addList(isRefresh: Boolean, videoBeen: List<VideoBean>)
  }

  interface Presenter : BasePresenter {
    fun getList(type: String, isRefresh: Boolean)
  }
}
